gwcs/wcs.py

```diff
@@ -634,3 +641,211 @@ def fix_inputs(self, fixed):
         new_pipeline.append((step0[0], new_transform))
         new_pipeline.extend(self.pipeline[1:])
         return self.__class__(new_pipeline)
+
+    def to_fits_sip(self, bounding_box, max_error=None, max_rms=None, degree=None,
+                    max_inv_error=None, max_inv_rms=None, verbose=False):
+        """
+        Construct a SIP-based approximation to the WCS in the form of a FITS header
+
+        This assumes a tangent projection.
+
+        Parameters
+        ----------
+        bounding_box : a pair of tuples, each consisting of two integers
+            Represents the range of pixel values in both dimensions
+            ((xmin, xmax), (ymin, ymax))
+        max_error : float
+            Maximum allowed error over the domain of the pixel array.
+        degree : int
+            Degree of the SIP polynomial. If supplied, max_error must be None.
+        max_inv_error : float
+            Maximum allowed inverse error over the domain of the pixel array.
+        verbose : bool
+            print progress of fits
+
+        Returns
+        -------
+        FITS header with all SIP WCS keywords
+
+        Raises
+        ------
+        ValueError
+            If the WCS is not 2D, an exception will be raised. If the specified accuracy
+            (both forward and inverse, both rms and maximum) is not achieved an exception
+            will be raised.
+
+        Notes
+        -----
+
+        Use of this requires a judicious choice of required accuracies. Attempts to use
+        higher degrees (~7 or higher) will typically fail due floating point problems
+        that arise with high powers.
+
+
+        """
+        if (self.forward_transform.n_inputs !=2
+            or self.forward_transform.n_outputs != 2):
+           raise ValueError("The to_fits_sip method only works with 2-dimensional transforms")
+
+        # Handle the options related to degree and thresholds.
+        if degree is not None and max_error is not None:
+            raise ValueError(
+                "Degree cannot be specified if max_error has a value")
+        if degree is None and max_error is None and max_rms is None:
+            raise ValueError(
+                "max_error must be specified if degree is None")
+        fits_dict = {}
+        transform = self.forward_transform
+        # Determine reference points.
+        (xmin, xmax), (ymin, ymax) = bounding_box
+        crpix1 = int((xmax - xmin) / 2)
+        crpix2 = int((ymax - ymin) / 2)
+        crval1, crval2 = transform(crpix1, crpix2)
+        hdr = fits.Header()
+        hdr['naxis'] = 2
+        hdr['naxis1'] = xmax
+        hdr['naxis2'] = ymax
+        hdr['ctype1'] = 'RA---TAN-SIP'
+        hdr['ctype2'] = 'DEC--TAN-SIP'
```

Comment:
I think these two lines setting `ctype` could be moved to the first branch of `if pdegree > 1` below. No need to assign `ctype` only to have it overwritten in the `else` branch.